toolkit/tools/internal/userutils/userutils.go

```diff
@@ -70,14 +70,20 @@ func UserExists(username string, installChroot safechroot.ChrootInterface) (bool
 	return userExists, nil
 }
 
-func AddUser(username string, hashedPassword string, uid string, installChroot safechroot.ChrootInterface) error {
+func AddUser(username string, homeDir string, primaryGroup string, hashedPassword string, uid string, installChroot *safechroot.Chroot) error {
 	var args = []string{username, "-m"}
 	if hashedPassword != "" {
 		args = append(args, "-p", hashedPassword)
 	}
 	if uid != "" {
 		args = append(args, "-u", uid)
 	}
+	if homeDir != "" {
+		args = append(args, "-d", homeDir)
+	}
+	if primaryGroup != "" {
```

Comment:
I think for non-rootfs we won't have created this group yet right? I'm thinking maybe `addGroups()` should take both the `config.Groups` and each users' custom group if set. It would need to be run for rootfs images as well though.

Reply:
No, considering [this PR](https://github.com/microsoft/CBL-Mariner/pull/6957) that is related to the changes in current PR, the groups will be created before adding the users for both rootfs and non-rootfs.

Reply:
Sorry, wasn't clear. I'm mostly worried about the case where a user might have a different group than the ones defined in `config.Groups`. Is this a possibility? My reading of useradd's docs suggest that `-g` only works on already existing groups. We can handle this two ways, either update `addGroups()`, or add a validation step to the config that enforces each user can only belong to groups that are defined.